src/Evolution/Systems/Cce/InitializeCce.cpp

```diff
@@ -0,0 +1,44 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Evolution/Systems/Cce/InitializeCce.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/SpinWeighted.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/Gsl.hpp"
+
+namespace Cce {
+
+void InitializeJ::apply(
+    const gsl::not_null<Scalar<SpinWeighted<ComplexDataVector, 2>>*> j,
+    const Scalar<SpinWeighted<ComplexDataVector, 2>>& boundary_j,
+    const Scalar<SpinWeighted<ComplexDataVector, 2>>& boundary_dr_j,
+    const Scalar<SpinWeighted<ComplexDataVector, 0>>& r) noexcept {
+  const size_t number_of_radial_points =
+      get(*j).size() / get(boundary_j).size();
+  const auto& one_minus_y_collocation =
+      1.0 - Spectral::collocation_points<Spectral::Basis::Legendre,
+                                         Spectral::Quadrature::GaussLobatto>(
+                number_of_radial_points);
+  for (size_t i = 0; i < number_of_radial_points; i++) {
+    ComplexDataVector angular_view_j{
+        get(*j).data().data() + get(boundary_j).size() * i,
+        get(boundary_j).size()};
+    const auto one_minus_y_coefficient =
```

Comment:
maybe add a comment that it is okay to use auto for these two expressions as they are used only once below